rpcs3/Emu/RSX/RSXFIFO.cpp

```diff
@@ -477,20 +477,19 @@ namespace rsx
 			fmt::throw_exception("Unexpected command 0x%x" HERE, cmd);
 		}
 
-		if (performance_counters.state != FIFO_state::running)
+		if (const auto state = std::exchange(performance_counters.state, FIFO_state::running);
+			state != FIFO_state::running)
 		{
-			//Update performance counters with time spent in idle mode
-			performance_counters.idle_time += (get_system_time() - performance_counters.FIFO_idle_timestamp);
-
-			if (performance_counters.state == FIFO_state::spinning)
+			// Hack: Delay FIFO wake-up according to setting
+			// NOTE: The typical spin setup is a NOP followed by a jump-to-self
+			// NOTE: There is a small delay when the jump address is dynamically edited by cell
+			if (state != FIFO_state::nop)
 			{
-				//TODO: Properly simulate FIFO wake delay.
-				//NOTE: The typical spin setup is a NOP followed by a jump-to-self
-				//NOTE: There is a small delay when the jump address is dynamically edited by cell
-				busy_wait(3000);
+				fifo_wake_delay();
 			}
 
-			performance_counters.state = FIFO_state::running;
+			//Update performance counters with time spent in idle mode
```

Comment:
We're already moving the comment, might as well update it with new format rules (space after //)